src/sync.cpp

```diff
@@ -36,6 +36,401 @@ const int Sync::FILE_UPDATE_DELAY_DS = 30;
 const int Sync::FILE_UPDATE_MAX_DELAY_SECS = 60;
 const dstime Sync::RECENT_VERSION_INTERVAL_SECS = 10800;
 
+namespace {
+
+// Collects all syncable filesystem paths in the given folder under `localpath`
+set<string> collectAllPathsInFolder(Sync& sync, MegaApp& app, FileSystemAccess& fsaccess, string localpath,
+                                    const string& localdebris, const string& localseparator, const bool followsymlinks)
+{
+    auto fa = std::unique_ptr<FileAccess>{fsaccess.newfileaccess()};
```

Comment:
This and others newfileaccess calls should pass false to followSymLinks. Otherwise we would be collecting data for the wrong files. As a matter of fact, if fa->mIsSymLink evals true, we shouldn't add the path until symlinks are supported

Reply:
This should be addressed now. Fun fact: You can't actually create links on FAT :)
Still good that we've got this part covered I think. Who knows, maybe we'll use the functionality for something else in the future, though I hope we don't have to...